jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4301,31 +4306,92 @@ ConstantPool getConstantPool() {
  *
  * @return		the host class of the receiver.
  */
-public Class<?> getNestHost() {
-	return getNestHostImpl();
+@CallerSensitive
+public Class<?> getNestHost() throws SecurityException {
+	if (null == nestHost) {
+		nestHost = getNestHostImpl();
+
+		/* The specification requires that if:
+		 *    - the returned class is not the current class
+		 *    - a security manager is present
+		 *    - the caller's class loader is not the same or an ancestor of the returned class
+		 *    - s.checkPackageAccess() disallows access to the package of the returned class
+		 * then throw a SecurityException.
+		 */
+		if (nestHost != this) {
+			SecurityManager securityManager = System.getSecurityManager();
+			if (nestHost == null) {
+				nestHost = this;
+			} else if (securityManager != null) {
+				ClassLoader callerClassLoader = ClassLoader.getCallerClassLoader();
+				ClassLoader nestHostClassLoader = nestHost.getClassLoader();			
+				if (!com.ibm.oti.util.Util.doesClassLoaderDescendFrom(nestHostClassLoader, callerClassLoader)) {
```

Comment:
you can - `import static com.ibm.oti.util.Util.doesClassLoaderDescendFrom;`. then, you can directly call `doesClassLoaderDescendFrom`.